### PR TITLE
General improvements in example app

### PR DESCRIPTION
**Pull request by tutok**
### Please check if the PR fulfills these requirements ###

- [x] The commit message follows our guidelines
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Changelog have been added / updated (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

**What kind of change does this PR introduce? (Bug fix, feature, docs update, ...)**

**What is the current behavior? (You can also link to an open issue here)**

**What is the new behavior (if this is a feature change)?**

**Does this PR introduce a breaking change? (What changes might users need to make in their application due to this PR?)**

**Other information:**

### CODE REVIEW

**Comment by tutok** [comment]
For now, I do not have a clear solution for rethrowing custom errors, looks like `componentDidCatch` flatten custom errors.

File: examples/shop-with-blog/client/src/pages/shop/Account/PersonalInformation.js [hunk 1]
```diff
@@ -1,57 +1,37 @@
...
+            {({ isSubmitting, status = {} }) => (
+              <GridLayout as={Form} id="edit-customer" i18nId="editCustomer" gridArea={TwoColumnsLayoutArea.left}>
+                <FormField name="firstname" required />
+                <FormField name="lastname" required />
+                <FormField name="email" type="email" required />
+                <FlexLayout justifyContent="space-between" alignItems="center" mt="md">
+                  <Link as={RouterLink} to="/account/change-password">
+                    <T id="editCustomer.changePassword" />
+                  </Link>
+                  <Button type="submit" variant={isSubmitting ? 'loader' : undefined}>
```
Comment (nimobeeren) [inline examples/shop-with-blog/client/src/pages/shop/Account/PersonalInformation.js]: If we merge #548 first we can replace this with `<FormSubmit />`

File: packages/falcon-shop-extension/src/types.ts [hunk 2]
```diff
@@ -67,14 +67,14 @@ export type EditAddressInput = AddressBase & {
 };
 
 export type Customer = {
-  id?: number;
-  websiteId?: number;
+  id: number;
+  websiteId: number;
   addresses?: Address[];
```
Comment (nimobeeren) [inline packages/falcon-shop-extension/src/types.ts]:
```suggestion
  addresses: Address[];
```
To match schema

File: examples/shop-with-blog/client/src/pages/shop/Account/ChangePassword.js [hunk 3]
```diff
@@ -1,65 +1,44 @@
...
-              </GridLayout>
-            )}
-          </Formik>
+      <ChangePasswordFormProvider onSuccess={() => history.push('/account')}>
+        {({ isSubmitting, status = {} }) => (
+          <Form id="change-password" i18nId="changePassword" gridArea={TwoColumnsLayoutArea.left}>
+            <FormField name="currentPassword" type="password" required validate={[]}>
+              {({ field }) => <PasswordRevealInput {...field} />}
+            </FormField>
+            <FormField name="password" type="password" required autoComplete="new-password">
```
[the remark this answers is not in the document]
Comment (tutok) [reply examples/shop-with-blog/client/src/pages/shop/Account/ChangePassword.js]: I think we should check if new password meets validation criteria https://github.com/deity-io/falcon/blob/db40dc1c61fc17eb3276cdeb7a46c6ba77337314/packages/falcon-front-kit/src/Forms/validators.ts#L24
So we should not override default validators here.

File: examples/shop-with-blog/client/src/pages/shop/Account/PersonalInformation.js (hunk 1 above)
Comment (nimobeeren) [reply examples/shop-with-blog/client/src/pages/shop/Account/PersonalInformation.js]: #548 has been merged. Could you try this change?

File: examples/shop-with-blog/client/src/pages/shop/Account/ChangePassword.js [hunk 4]
```diff
@@ -1,65 +1,44 @@
...
+              {({ field }) => <PasswordRevealInput {...field} />}
+            </FormField>
+            <FlexLayout justifyContent="flex-end" alignItems="center" mt="md">
+              <Button as={NavLink} to="/account/personal-information" mr="md">
+                <T id="changePassword.cancelButton" />
+              </Button>
+              <Button type="submit" variant={isSubmitting ? 'loader' : undefined}>
```
Comment (nimobeeren) [inline examples/shop-with-blog/client/src/pages/shop/Account/ChangePassword.js]: Here we can also use `<FormSubmit />`

Comment (tutok) [reply examples/shop-with-blog/client/src/pages/shop/Account/ChangePassword.js]: done

**Pull request merged**